Monero-Peer-Subnet-Deduplication/code/Rust/src/main.rs

```diff
@@ -51,7 +63,7 @@ static CONNECTOR: OnceLock<
 static BAD_PEERS_CHANNEL: OnceLock<mpsc::Sender<(SocketAddr, bool)>> = OnceLock::new();
 
 /// A [`Semaphore`] to limit the amount of concurrent connection attempts so we don't overrun ourself.
-static CONNECTION_SEMAPHORE: Semaphore = Semaphore::const_new(100);
+static CONNECTION_SEMAPHORE: Semaphore = Semaphore::const_new(1);
```

Comment:
If you make this a `OnceLock` you can change the number of permits at runtime: 

```suggestion
static CONNECTION_SEMAPHORE: OnceLock<Semaphore> = OnceLock::new();
```

Then somewhere in `main` add the code to set the `Semaphore`:

```rust
CONNECTION_SEMAPHORE.set(Semaphore::new(X));
```

Then on the usages add `.get().unwrap()`